src/vs/workbench/contrib/chat/browser/chatContentParts/toolInvocationParts/chatTerminalToolProgressPart.ts

```diff
@@ -943,16 +891,37 @@ class ChatTerminalToolOutputSection extends Disposable {
 		this._outputScrollbar.setScrollPosition({ scrollTop: dimensions.scrollHeight });
 	}
 
-	private _getOutputContentHeight(): number {
-		const firstChild = this._outputBody.firstElementChild as HTMLElement | null;
-		if (!firstChild) {
-			return this._outputBody.scrollHeight;
-		}
+	private _getOutputContentHeight(lineCount?: number): number {
 		const style = dom.getComputedStyle(this._outputBody);
 		const paddingTop = Number.parseFloat(style.paddingTop || '0');
 		const paddingBottom = Number.parseFloat(style.paddingBottom || '0');
 		const padding = paddingTop + paddingBottom;
-		return firstChild.scrollHeight + padding;
+		const contentHeight = this._calculateContentHeight(lineCount);
+		return contentHeight + padding;
+	}
+
+	private _calculateContentHeight(lineCount?: number): number {
+		if (lineCount === 0 || lineCount === undefined) {
+			return MIN_OUTPUT_HEIGHT;
+		}
+		const rowHeightPx = this._computeRowHeightPx();
+		return lineCount * rowHeightPx;
+	}
+
+	private _computeRowHeightPx(): number {
```

Comment:
Would it be possible to just do away with this px calculation completely if we allowed up to x rows instead regardless of the actual pixel size?